torch/distributed/tensor/_ops/_pointwise_ops.py

```diff
@@ -427,22 +427,60 @@
 # Ops that preserve specific Partial types through the operation.
 # For example, torch.maximum preserves Partial("max") because
 # max(max(a), max(b)) == max(a, b).
-partial_preserving_ops: dict[torch._ops.OpOverload, str] = {
-    aten.maximum.default: "max",
-    aten.maximum.out: "max",
-    aten.minimum.default: "min",
-    aten.minimum.out: "min",
+partial_preserving_ops: dict[torch._ops.OpOverload, list[str]] = {
+    aten.maximum.default: ["max"],
+    aten.maximum.out: ["max"],
+    aten.minimum.default: ["min"],
+    aten.minimum.out: ["min"],
 }
 
 
+def _find_last_non_matching_partial_index(
+    arg_strategy: OpStrategy,
+    preserve_partial_types: list[str],
+) -> int:
+    """
+    Returns the minimum "last non-matching partial index" across all OpSpecs.
+
+    For each OpSpec, finds the index of the last Partial that doesn't match
+    preserve_partial_types. Takes the min across all OpSpecs (best case).
```

Comment:
an example would help here.

Also, you might be the first person to talk about 'Partial order' so it bears writing something explicitly about what your assumptions are here.  We are allowing shards to be in any order (using ShardOrder) but are we assuming partials are always created in L2R order?  How does DTensor currently control the creation order of Partials?